src/main/java/uk/gov/hmcts/probate/service/NotificationService.java

```
@@ -973,4 +980,66 @@ private List<String> getExecutorsNamesList(List<CollectionMember<ExecutorApplyin
+            final String message = MessageFormat.format(
+                    "Unable to parse grant issued date: [{}] (expecting yyyy-mm-dd format)",
+                    grantIssuedCase);
+            throw new BusinessValidationException(message, e.getMessage());
+        }
+        final String grantIssuedOn = caseData.getGrantIssuedDateFormatted();
+        final String grantIssuedOnCy = localDateToWelshStringConverter.convert(grantIssuedDate);
+
+        final String addresseeName = switch (caseData.getApplicationType()) {
+            case PERSONAL -> caseData.getPrimaryApplicantFullName();
+            case SOLICITOR -> caseData.getSolsSOTName();
+        };
+
+        final Map<String, Object> personalisation = Map.of(
+                "ccd_reference", caseRef,
+                "deceased_name", deceasedName,
+                "deceased_dod", deceasedDiedOn,
+                "deceased_dod_cy", deceasedDiedOnCy,
+                PERSONALISATION_APPLICANT_NAME, addresseeName,
+                "grant_issued_date", grantIssuedOn,
+                "grant_issued_date_cy", grantIssuedOnCy);
+
+        try {
+            final SendEmailResponse response = notificationClientService.sendEmail(
+                    templateId,
+                    recipientEmail,
+                    personalisation,
+                    caseRef);
+            log.info("Sent notification for move to Post Grant Issued for case: {}", caseRef);
+            final Document sentEmail = getGeneratedSentEmailDocument(
+                    response,
+                    recipientEmail,
+                    SENT_EMAIL);
+            log.info("Got PDF of Post Grant Issued notification for case: {}", caseRef);
+            return sentEmail;
+        } catch (NotificationClientException e) {
+            log.info("Failed to send Post Grant Issued notification for case {}, message: {}", caseRef, e.getMessage());
+            return null;
+        }
```

if the email was sent successfully but failed to upload, or an error occurred during **getGeneratedSentEmailDocument**, the exception is not handled here.

---

i've updated this to explicitly catch and log any runtimeexception in the document generation / upload process